src/Control/Distributed/Process/Internal/Primitives.hs

```diff
@@ -400,8 +400,10 @@ newtype Match b = Match { unMatch :: MatchOn Message (Process b) }
 receiveWait :: [Match b] -> Process b
 receiveWait ms = do
   queue <- processQueue <$> ask
-  Just proc <- liftIO $ dequeue queue Blocking (map unMatch ms)
-  proc
+  mProc <- liftIO $ dequeue queue Blocking (map unMatch ms)
+  case mProc of
+    Just proc' -> proc'
+    Nothing    -> error "Well... That wasn't supposed to happen! o_O"
```

Comment:
It's better to elaborate on this error a bit more so if for some reason (that may appear in the future this error will be shown, a user will not be surprised.
In general, it makes sense to have an error or better `die` here, because a type of the dequeue is not precise enough to tell us that this case is not possible.